include/rapidcheck/detail/Utility.h

```diff
@@ -95,7 +95,16 @@ inline uint64_t avalanche(uint64_t x) {
 /// and the rest set to 0.
 template <typename T>
 constexpr T bitMask(int nbits) {
-  return ~((~static_cast<T>(0) - 1) << static_cast<T>(nbits - 1));
+  using UT = typename std::make_unsigned<T>::type;
+  // There are two pieces of undefined behavior we're avoiding here,
+  //   1. Shifting past the width of a type (ex `<< 32` against an `int32_t`)
+  //   2. Shifting a negative operand (which `~0` is for all signed types)
+  // First we branch to avoid shifting the bit-width of the given type, then
+  // (assuming we are shifting, and aren't just returning all 1s) we cast `~0`
+  // to an explicitly signed type before performing the shift.
+  return nbits != sizeof(T) * 8                            ?
+         static_cast<T>(~(static_cast<UT>(~(0)) << nbits)) :
+         static_cast<T>(~(0));
```

Comment:
Shouldn't this be `~static_cast<T>(0)` or even shorter, `~T(0)`? That is, casting before complementing. Same for the line above, shouldn't that be `~UT(0)`?